lib/extensions/ar_taggable.rb

```diff
@@ -77,6 +77,38 @@ def tags(options = {})
       options[:ns] = Tag.get_namespace(options)
       Tag.tags(options)
     end
+
+    # defines an attribute that detects a certain tag
+    # define has_attrs, has_attrs? and attr_tags
+    # arel returns true or nil (false)
+    #   a count() > 0 or a case statement could return true, false (instead of null) but this seemed the least verbose
+    def tag_attribute(attribute_name, ns)
+      plural_attribute_name = "has_#{attribute_name.to_s.pluralize}"
+
+      virtual_attribute plural_attribute_name, :boolean, :uses => :tags, :arel => (lambda do |t|
+        ta = Tag.arel_table
+        tnga = Tagging.arel_table
+        t.grouping(
+          Arel.sql(
+            Tagging.joins(:tag).select('true')
+                   .where(ta[:name].matches("#{ns}/%", nil, true))
+                   .where(tnga[:taggable_type].eq(base_class.name).and(tnga[:taggable_id].eq(arel_table[:id])))
+                   .limit(1).to_sql
+          )
+        )
+      end)
+
+      define_method("#{attribute_name}_tags") do
+        Tag.filter_ns(tags, ns)
+      end
+
+      define_method(plural_attribute_name) do
+        has_attribute?(plural_attribute_name) ? self[plural_attribute_name] || false : Tag.filter_ns(tags, ns).size > 0
```

Comment:
Was going to say "why not use the method that you are defining above for this" (or "make it DRY"), but realized that it probably would be more characters to do that.

That said, this is probably a case where a ternary is not a good use case, and you should just expand this to a multi-line `if`:

```ruby
if has_attribute?(plural_attribute_name)
  self[plural_attribute_name] || false
else
  Tag.filter_ns(tags, ns).size > 0
end
```

Reply:
Heh, apparently @Fryguy and myself were thinking the same thing:

https://github.com/ManageIQ/manageiq/pull/20381#discussion_r459627029

Reply:
I commented over there, but generally a attribute accessor method is expected to under the covers use `read_attribute`

Reply:
Think we're on the same page.

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods.rb#L322

```ruby
def [](attr_name)
  read_attribute(attr_name) { |n| missing_attribute(n, caller) }
end
```

Reply:
Oh nice... This works for me, then.

Reply:
I liked changing over to `read_attribute` - they are "the same thing" but using the actual phrase `read_attribute` did lower the cognitive load for this (for some reason) - It was easier to go with the flow than ask why I found it easier to read.

I was reluctant at first, but I like the suggestion @Fryguy (and Nick having issues with the same method didn't hurt the matter either. thnx @NickLaMuro )